day1/es1/SwitchButton.qml

```diff
@@ -0,0 +1,32 @@
+import QtQuick 2.0
+
+Row {
+    id: switchButton
+
+    width: 160
+    height: 40
+    anchors.centerIn: parent
+    state: "unchecked"
```

Comment:
Gli stati sono molti potenti in QML, generalmente si usano per definire delle "macchine a stati", e talvolta possono risultare anche un po' fragili visto che si basano sulle stringhe: un piccolo typo in una stringa ed ecco che il meccanismo di controllo dello stato che stai usando potrebbe rompersi.

Per casi così semplici dove le alternative sono due, solitamente si usano delle properti booleane.
Puoi definire una property booleana con:
`property bool nome_property`

E' buona norma dare un nome facilmente leggibile alle property booleane (spesso iniziano col verbo inglese `is`), ad esempio `isActive`.

Proviamo a usare una property booleana, vedrai che rimuoviamo diverse righe di codice! :D

Reply:
Questo si può anche marchiare come risolto: l'esercizio richiedeva esplicitamente di usare gli `State`.

Reply:
usati PropertyChanges per modificare i colori

Reply:
Perfetto, si può risolvere